pkg/morph/client/notifications.go

```diff
@@ -226,6 +226,7 @@ func (c *Client) restoreSubscriptions(cli *rpcclient.WSClient, endpoint string)
 
 	// notification events restoration
 	for contract := range c.subscribedEvents {
+		contract := contract
 		id, err = cli.SubscribeForExecutionNotifications(&contract, nil)
```

Comment:
Maybe this would be better handled in WSClient itself.

Reply:
nspcc-dev/neo-go#2890

Reply:
Agree, but I still believe it is a bug: `SubscribeForExecutionNotifications` gives no guarantees (at least they are not obvious) and it is an external library which is harder to audit during our code reviews.
Right now it is also faster to just fix it here.

Do you think it is a bug in `WSClient` and we need to create an issue for neo-go?

Reply:
Already there. I agree that pointer-level guarantees are very vague in Go and of course this should be fixed in the node right away, without waiting for NeoGo fix. But it's too easy to make a mistake like this, so WSClient should be fixed (we're very likely to see similar issues with other subscribers otherwise).